src/rdkafka_broker.c

```diff
@@ -2194,14 +2268,14 @@ rd_kafka_broker_handle_ApiVersion (rd_kafka_t *rk,
 
 
 	if (err) {
-		rd_kafka_broker_fail(rkb, LOG_DEBUG,
-				     RD_KAFKA_RESP_ERR__TRANSPORT,
-				     "ApiVersionRequest failed: %s: "
-				     "probably due to incorrect "
-                                     "security.protocol "
-                                     "or broker version is < 0.10 "
-                                     "(see api.version.request)",
-				     rd_kafka_err2str(err));
+                if (rkb->rkb_transport)
+                        rd_kafka_broker_fail(
+                                rkb, LOG_WARNING,
+                                RD_KAFKA_RESP_ERR__TRANSPORT,
+                                "ApiVersionRequest failed: %s: "
+                                "probably due to broker version < 0.10 "
```

Comment:
Should we mention the possibility of a protocol mismatch here as well?

Reply:
We should not arrive here on Disconnect, so those cases are handled separately.